cache/file/file.go

```diff
@@ -0,0 +1,52 @@
+// Package file provides a disk based cache.
+package file
+
+import (
+	"github.com/pierrre/imageserver"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+// Cache is a implementation of disk based cache system.
+type Cache struct {
+	Path string
+}
+
+// Get implements imageserver/cache.Cache.
+func (cache *Cache) Get(key string, params imageserver.Params) (*imageserver.Image, error) {
+	data := cache.getData(key)
+	if data == nil {
+		return nil, nil
+	}
+	im := new(imageserver.Image)
+	err := im.UnmarshalBinaryNoCopy(data)
+	if err != nil {
+		return nil, err
+	}
+	return im, nil
+}
+
+func (cache *Cache) getData(key string) []byte {
+	item, err := ioutil.ReadFile(cache.Path + "/" + key)
+	if err != nil {
+		return nil
+	}
+	return item
+}
+
+// Set implements imageserver/cache.Cache.
+func (cache *Cache) Set(key string, im *imageserver.Image, params imageserver.Params) error {
+	data, err := im.MarshalBinary()
+	if err != nil {
+		return err
+	}
+	return cache.setData(key, data)
+}
+
+func (cache *Cache) setData(key string, data []byte) error {
+	mutex := new(sync.Mutex)
+	mutex.Lock()
+	defer mutex.Unlock()
+	return ioutil.WriteFile(cache.Path+"/"+key, data, os.ModePerm)
```

Comment:
Please don't use `os.ModePerm` (0777).
Use 0644 instead.

If you want to give the ability to customize the value to the user, you should add a variable to `Cache`. (Don't do it if you think that 0644 is fine)